packages/nimble-components/src/rich-text-viewer/index.ts

```diff
@@ -11,7 +18,99 @@ declare global {
 /**
  * A nimble styled rich text viewer
  */
-export class RichTextViewer extends FoundationElement {}
+export class RichTextViewer extends FoundationElement {
+    /**
+     *
+     * @public
+     * Markdown string to render its corresponding rich text content in the component.
+     */
+    @observable
+    public markdown = '';
+
+    /**
+     * @internal
+     */
+    public viewer!: HTMLDivElement;
+    private readonly markdownParser: MarkdownParser;
+    private readonly domSerializer: DOMSerializer;
+
+    public constructor() {
+        super();
+        this.domSerializer = DOMSerializer.fromSchema(schema);
+        this.markdownParser = this.initializeMarkdownParser();
+    }
+
+    /**
+     * @internal
+     */
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.updateView();
+    }
+
+    /**
+     * @internal
+     */
+    public markdownChanged(): void {
+        if (this.$fastController.isConnected) {
+            this.updateView();
+        }
+    }
+
+    private initializeMarkdownParser(): MarkdownParser {
+        /**
+         * It configures the tokenizer of the default Markdown parser with the 'zero' preset.
+         * The 'zero' preset is a configuration with no rules enabled by default to selectively enable specific rules.
+         * https://github.com/markdown-it/markdown-it/blob/2b6cac25823af011ff3bc7628bc9b06e483c5a08/lib/presets/zero.js#L1
+         *
+         */
+        const zeroTokenizerConfiguration = defaultMarkdownParser.tokenizer.configure('zero');
+
+        // The detailed information of the supported rules were provided in the below CommonMark spec document.
+        // https://spec.commonmark.org/0.30/
+        const supportedTokenizerRules = zeroTokenizerConfiguration.enable([
+            'emphasis',
+            'list',
+            'autolink'
+        ]);
+
+        // Disabling normalize which replaces the null character('\0') with the Unicode representation('\uFFFD') of Replacement Character(�)
+        // https://github.com/markdown-it/markdown-it/blob/2b6cac25823af011ff3bc7628bc9b06e483c5a08/lib/rules_core/normalize.js#L15
+        supportedTokenizerRules.disable('normalize');
```

Comment:
In [one of the `wackyStrings` test cases](https://github.com/ni/nimble/pull/1335/files#diff-02e8008d23a4fd737b037e2232c4160dca3e145bf526690b41f204f626563841R527), we are checking the [`\x00` character ](https://github.com/ni/nimble/blob/e6e254dd674d70ca392f598e1073df718455a00b/packages/nimble-components/src/utilities/tests/wacky-strings.ts#L9)and the expectation for the other `wackyStrings` is that it should render the same text within a `<p>` tag which passes as an input. But for `\x00`, it replaces and renders as `�`. So to stay consistent behavior with other strings I have disabled the `normalize`.

But rendering the `�` for the null character is the expected behavior I could update the test case accordingly.

Reply:
I wouldn't let the wackyStrings test cases drive our behavior. They're great for helping us discover cases we wouldn't otherwise think of but we don't necessarily have to treat all those cases the same.

My vote would be to re-enable normalization. I think it's likely that users would want to know about strange characters in the text (for example, a lot of code editors started supporting a way to visualize control characters in the past years because they were being used as an attack vector).

If we do that you could update the wacky strings test to skip or special case the strings that no longer pass.

Reply:
when you do it please use a pattern that clearly filters the list in one test and then add an additional test for the filtered case. (don't conditionally run different expect statements in the same test)

Reply:
Re-enabled normalization and updated the test cases as suggested.